src/ui_fsmenu/mapdetailsbox.cc

```diff
@@ -0,0 +1,178 @@
+
+#include "graphic/image_cache.h"
+#include "map_io/map_loader.h"
+
+MapDetailsBox::MapDetailsBox(
+   Panel* parent, uint32_t, uint32_t standard_element_height, uint32_t padding, int32_t, int32_t)
+   : UI::Box(parent, 0, 0, UI::Box::Vertical),
+     title_(this,
+            0,
+            0,
+            0,
+            0,
+            _("Map"),
+            UI::Align::kCenter,
+            g_style_manager->font_style(UI::FontStyle::kFsGameSetupHeadings)),
+     title_box_(this, 0, 0, UI::Box::Horizontal),
+     content_box_(this, 0, 0, UI::Box::Vertical),
+     map_name_(&title_box_,
+               0,
+               0,
+               0,
+               0,
+               _("No map selected"),
+               UI::Align::kLeft,
+               g_style_manager->font_style(UI::FontStyle::kLabel)),
+     select_map_(&title_box_,
+                 "change_map_or_save",
+                 0,
+                 0,
+                 standard_element_height,
+                 standard_element_height,
+                 UI::ButtonStyle::kFsMenuSecondary,
+                 g_image_cache->get("images/wui/menus/toggle_minimap.png"),
+                 _("Change map or saved game")),
+     map_description_(&content_box_,
+                      0,
+                      0,
+                      UI::Scrollbar::kSize,  // min width must be set to avoid assertion failure...
+                      0,
+                      UI::PanelStyle::kFsMenu,
+                      "",
+                      UI::Align::kLeft,
+                      UI::MultilineTextarea::ScrollMode::kNoScrolling),
+     suggested_teams_box_(&content_box_, 0, 0, UI::Box::Vertical, 4, 0, 0, 0) {
+	content_box_.set_scrolling(true);
+	add(&title_, Resizing::kAlign, UI::Align::kCenter);
+	add_space(3 * padding);
+	title_box_.add(&map_name_, UI::Box::Resizing::kAlign, UI::Align::kLeft);
+	title_box_.add_inf_space();
+	title_box_.add(&select_map_, UI::Box::Resizing::kAlign, UI::Align::kRight);
+	add(&title_box_, UI::Box::Resizing::kFullSize);
+	add_space(3 * padding);
+	add(&content_box_, UI::Box::Resizing::kExpandBoth);
+	content_box_.add(&map_description_, UI::Box::Resizing::kExpandBoth);
+	content_box_.add_space(3 * padding);
+	content_box_.add(&suggested_teams_box_, UI::Box::Resizing::kExpandBoth);
+}
+MapDetailsBox::~MapDetailsBox() {
+}
+
+void MapDetailsBox::update_from_savegame(GameSettingsProvider* settings) {
+	const GameSettings& game_settings = settings->settings();
+
+	select_map_.set_visible(settings->can_change_map());
+	select_map_.set_enabled(settings->can_change_map());
+
+	show_map_description_savegame(game_settings);
+	show_map_name(game_settings);
+}
+void MapDetailsBox::show_map_description_savegame(const GameSettings& game_settings) {
+	std::string infotext = _("Saved players are:");
+
+	for (uint8_t i = 0; i < game_settings.players.size(); ++i) {
+		const PlayerSettings& current_player = game_settings.players.at(i);
+		infotext += "\n* ";
```

Comment:
I imagine this kind of piecewise assembly would break with RTL languages. Please check whether you can use multiple nested `boost::format(_("..."))` instead. Below as well.

Reply:
hmm, will check how to do that...
actually this is code as it was - just moved it into a new class

Reply:
Fair point. Do you want to look into this here? If not, just leave a TODO and I'll have some fun with it after this branch is merged

Reply:
I'll leave a TODO, invested enough time in this one already... ;)